### DISCUSSION TITLE
How best to target sub-arrays for iteration?

### DISCUSSION THREAD

**Discussion by Truncated** [farling42/obsidian-import-json#73] [Q&A] [answered]
[Continuing the workflow process for the last Q&A](https://github.com/farling42/obsidian-import-json/discussions/72#discussion-6570022), I'm hoping I'm just missing something.

Given the schema below, is there a way I can target the sub-arrays:
`editionTwoFutureLogPages, editionTwoIndexPages, indexPages, pages`
allowing for an import job to use those subfields as separate notes without hacking at the JSON to break the array apart?

<details>
<summary>Example Schema stripped down</summary>

```
{
    "libraryVolumes": [
        {
            "coverPage": {
                "created": "2023-10-31T12:12:59Z",
                "id": "0ccc874f-7f11-422d-bcb9-8cf9edefb64a",
                "image": {
                    "id": "fe17df79-feb7-4001-ae49-f9d3eda46822"
                }
            },
            "editionTwoFutureLogPages": [
                {
                    <<VALUES for Page 3>>
                },
                {
                    <<VALUES for Page 4>>
                }
            ],
            "editionTwoIndexPages": [
                {
                    <<VALUES for Page 1>>
                },
                {
                    <<VALUES for Page 2>>
                }
            ],
            "endDate": "2024-02",
            "firstPageSide": "left",
            "id": "f9c1a5ed-59bf-4a8a-ab03-e4f550ca8734",
            "indexPages": [],,
            "pages": []
        },
        {
            "coverPage": {
                "created": "2023-10-31T12:12:59Z",
                "id": "0ccc874f-7f11-422d-bcb9-8cf9edefb64a",
                "image": {
                    "id": "fe17df79-feb7-4001-ae49-f9d3eda46822"
                }
            },
            "editionTwoFutureLogPages": [],
            "editionTwoIndexPages": [],
            "endDate": "2024-02",
            "firstPageSide": "left",
            "id": "f9c1a5ed-59bf-4a8a-ab03-e4f550ca8734",
            "indexPages": [
                {
                    <<VALUES for Page 3>>
                },
                {
                    <<VALUES for Page 4>>
                }
            ],
            "pages": [
                {
                    <<VALUES for Page 1>>
                },
                {
                    <<VALUES for Page 2>>
                }
            ]
        },
        {
            "coverPage": {
                "created": "2023-10-31T12:12:59Z",
                "id": "0ccc874f-7f11-422d-bcb9-8cf9edefb64a",
                "image": {
                    "id": "fe17df79-feb7-4001-ae49-f9d3eda46822"
                }
            },
            "editionTwoFutureLogPages": [],
            "editionTwoIndexPages": [
                {
                    "created": "2024-04-13T16:52:46Z",
                    "id": "f05f9480-f9bd-4adc-adf8-74085c55b772",
                    "image": {
                        "id": "7adc229c-d0e0-4739-870b-1c0f607565a9"
                    },
                    "page": 4
                },
                {
                    "created": "2024-04-13T16:53:09Z",
                    "id": "6e7d37cc-d591-4336-8b5f-ba5e0ebb8cf2",
                    "image": {
                        "id": "b8341be4-4c86-4177-b8e9-e68158f04a80"
                    },
                    "page": 5
                }
            ],
            "endDate": "2024-02",
            "firstPageSide": "left",
            "id": "f9c1a5ed-59bf-4a8a-ab03-e4f550ca8734",
            "indexPages": [],
            "pages": []
        }
    ]
}
```

</details>

The only way I can think of getting this to work is to drop the JSON objects from libraryVolumes to the root, then import them by specifying the sub-array fields (editionTwoFutureLogPages, editionTwoIndexPages, indexPages, and pages) one at a time. :(
I could use comparison statements to build a single template that would be appropriate for all four, but I'm hoping there's some smoother method that I'm just not privvy to.

IF there's not, is it possible to identify multiple fields to have the importer cycle through them all in one pass if the same template can work for all of the fields?

**Comment by farling42** [comment, marked as answer]
There is no current way to be able to create notes from all the sub-arrays.

At the moment, it is only possible to handle a single key to cycle through - although it _might_ be a fairly enhancement to allow multiple keys to be used. There is an enhancement request (#30 ) currently open to add support for multiple passes to be made on the same data file with slightly different import parameters on each pass.

**Comment by Truncated** [comment]
Okay, I'll add my example theory there then. Cheers!

**Comment by farling42** [comment]
It has been a long while, but version 0.38.1 now supports creating notes from sub-arrays within higher-level arrays.

So you could set "Field containing the data" to `libraryVolumes[libvol].editionTwoFutureLogPages`  and within the MD file you can reference the cover page with `@libvol.coverPage.created`.

It isn't possible to process several different arrays directly, but the batch processing might be able to do each array separately for you.
